Code/GraphMol/RGroupDecomposition/RGroupGa.cpp

```diff
@@ -110,7 +110,7 @@ RGroupGa::RGroupGa(const RGroupDecompData& rGroupData,
     }
     chromosomePolicy.setMax(pos, m.size());
     unsigned long count = numPermutations * m.size();
-    numPermutations = count / m.size() == numPermutations
+    numPermutations = count < numeric_limits<unsigned int>::max() && count / m.size() == numPermutations
```

Comment:
if count == unsigned int::max, we effectively halve it and then double it which makes count == 0.  This causes issues downstream.

Reply:
I'm not really following the logic of either the original code or this change. Couldn't you achieve an equivalent result with:
```
numPermutations = std::min(count,numeric_limits<unsigned int>::max());
```

Reply:
I think the original code was a way to check for overflow.  I don’t know if the codes are equivalent.

Reply:
I believe the original code I wrote was to check for overflow.  It would make sense if both count and numPermutations were the same unsigned type.  Given count is unsigned long and numPermutations is unsigned int Greg's suggestion looks perfect.

Reply:
I verified that the suggestion works on my test sets.